impl/open_search.py

```diff
@@ -0,0 +1,226 @@
+from django.core.serializers import serialize
+from django.http import JsonResponse
+
+# import settings.settings
+from django.conf import settings
+from ezidapp.models.identifier import Identifier
+import json
+import pdb
+import base64
+import datetime
+import ezidapp.models.validation as validation
+import impl.util
+import requests
+from django.conf import settings
+from urllib.parse import quote
+import re
+import functools
+
+# the functools allows memoizing the results of functions so they're not recalculated every time (ie cached
+# results if called more than once on the same instance)
+
+MAX_SEARCHABLE_TARGET_LENGTH = 255
+INDEXED_PREFIX_LENGTH = 50
+
+# testing
+# python manage.py shell
+
+# seems like they're using the django db model libraries https://docs.djangoproject.com/en/5.0/topics/db/queries/
+
+"""
+-- basic testing
+import impl.open_search as os
```

Comment:
Maybe use a different shorthand name than `os` to avoid confusion with the operating system library.

Reply:
Changed the name.